flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -82,202 +82,12 @@
-
-		final SingleSlotTestingSlotOwner slotOwner = new SingleSlotTestingSlotOwner();
-
-		final LogicalSlot slot = createTestingLogicalSlot(slotOwner);
-
-		final LogicalSlot otherSlot = new TestingLogicalSlotBuilder().createTestingLogicalSlot();
-
-		CompletableFuture<Execution> allocationFuture = execution.allocateResourcesForExecution(
-			executionGraph.getSlotProviderStrategy(),
-			LocationPreferenceConstraint.ALL,
-			Collections.emptySet());
-
-		assertFalse(allocationFuture.isDone());
-
-		assertEquals(ExecutionState.SCHEDULED, execution.getState());
-
-		// assign a different resource to the execution
-		assertTrue(execution.tryAssignResource(otherSlot));
-
-		// completing now the future should cause the slot to be released
-		slotFuture.complete(slot);
-
-		assertEquals(slot, slotOwner.getReturnedSlotFuture().get());
-	}
-
 	private TestingLogicalSlot createTestingLogicalSlot(SlotOwner slotOwner) {
 		return new TestingLogicalSlotBuilder()
 			.setSlotOwner(slotOwner)
 			.createTestingLogicalSlot();
 	}
 
-	/**
-	 * Tests that the slot is released in case of a execution cancellation when having
-	 * a slot assigned and being in state SCHEDULED.
-	 */
-	@Test
-	public void testSlotReleaseOnExecutionCancellationInScheduled() throws Exception {
-		final JobVertex jobVertex = createNoOpJobVertex();
-		final JobVertexID jobVertexId = jobVertex.getID();
-
-		final SingleSlotTestingSlotOwner slotOwner = new SingleSlotTestingSlotOwner();
-
-		final LogicalSlot slot = createTestingLogicalSlot(slotOwner);
-
-		final ProgrammedSlotProvider slotProvider = new ProgrammedSlotProvider(1);
-		slotProvider.addSlot(jobVertexId, 0, CompletableFuture.completedFuture(slot));
-
-		ExecutionGraph executionGraph = ExecutionGraphTestUtils.createSimpleTestGraph(
-			slotProvider,
-			new NoRestartStrategy(),
-			jobVertex);
-
-		executionGraph.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-
-		ExecutionJobVertex executionJobVertex = executionGraph.getJobVertex(jobVertexId);
-
-		final Execution execution = executionJobVertex.getTaskVertices()[0].getCurrentExecutionAttempt();
-
-		CompletableFuture<Execution> allocationFuture = execution.allocateResourcesForExecution(
-			executionGraph.getSlotProviderStrategy(),
-			LocationPreferenceConstraint.ALL,
-			Collections.emptySet());
-
-		assertTrue(allocationFuture.isDone());
-
-		assertEquals(ExecutionState.SCHEDULED, execution.getState());
-
-		assertEquals(slot, execution.getAssignedResource());
-
-		// cancelling the execution should move it into state CANCELED
-		execution.cancel();
-		assertEquals(ExecutionState.CANCELED, execution.getState());
-
-		assertEquals(slot, slotOwner.getReturnedSlotFuture().get());
-	}
-
-	/**
-	 * Tests that the slot is released in case of a execution cancellation when being in state
-	 * RUNNING.
-	 */
-	@Test
-	public void testSlotReleaseOnExecutionCancellationInRunning() throws Exception {
```

Comment:
I think this test is also still valid because we want to return the logical slot to its owner if the `Execution` is cancelled.

Reply:
I think this test is not needed because 
- the slot releasing check is covered by `ExecutionTest#testCanceledExecutionReturnsSlot`
- the state transition check is covered in `ExecutionVertexCancelTest#testCancelFromRunning`

Reply:
Thanks for the information @zhuzhurk. I think you are right.